lib/app/view/app.dart

```diff
@@ -112,72 +92,75 @@ class _AppViewState extends State<AppView> {
         return DarkAcademia.getTheme(fontFamily);
       case Themes.monochromePink:
         return MonochromePink.getTheme(fontFamily);
-
       default:
         return CoralBubble.getTheme(fontFamily);
     }
   }
 
   @override
   Widget build(BuildContext context) {
-    return Builder(
-      builder: (context) {
-        final themeState = context.watch<ThemeCubit>().state;
-        final localeCubit = context.watch<LocaleCubit>().state;
-        final fontCubit = context.watch<FontCubit>().state;
-
-        return MaterialApp(
-          navigatorKey: _navigatorKey,
-          debugShowCheckedModeBanner: false,
-          title: "My Dairy",
-          locale: localeCubit.currentLocale,
-          supportedLocales: S.delegate.supportedLocales,
-          localizationsDelegates: const [
-            S.delegate,
-            GlobalMaterialLocalizations.delegate,
-            GlobalCupertinoLocalizations.delegate,
-            GlobalWidgetsLocalizations.delegate
-          ],
-          theme: getThemeData(themeState.theme, fontCubit.currentFontFamily),
-          builder: (BuildContext context, child) {
-            return BlocListener<AuthSessionBloc, AuthSessionState>(
-              listener: (context, state) {
-                log.d("Auth session state is $state");
-
-                //! Currently we are not passing lastLoggedinUser anywhere, if you implement session
-                //! Rememeber to pass lastloggedinuser id as parameter to AuthPage
-
-                if (state is Unauthenticated) {
-                  bool isPINLoginEnabled =
-                      sl<PINAuthRepository>().isPINAuthEnabled();
-
-                  if (isPINLoginEnabled == true) {
-                    _navigator.pushAndRemoveUntil(
-                        MaterialPageRoute(builder: (_) => const PINAuthPage()),
-                        (route) => false);
-                  } else {
-                    _navigator.pushAndRemoveUntil(
-                        MaterialPageRoute(builder: (_) => AuthPage()),
-                        (route) => false);
-                  }
-                } else if (state is Authenticated) {
-                  log.d("freshLogin = ${state.freshLogin}");
-
-                  if (state.freshLogin == true) {
-                    _navigator.pushAndRemoveUntil(
-                        MaterialPageRoute(builder: (_) => const HomePage()),
-                        (route) => false);
-                  } else {
-                    _navigator.pop();
-                  }
-                }
+    return BlocBuilder<ThemeCubit, ThemeState>(
+      builder: (context, themeState) {
+        return BlocBuilder<LocaleCubit, LocaleState>(
+          builder: (context, localeState) {
+            return BlocBuilder<FontCubit, FontState>(
+              builder: (context, fontState) {
+                return MaterialApp(
+                  navigatorKey: _navigatorKey,
```

Comment:
The performance gain may not be worth the nested widget BlocBuilders. These bloc's rarely change, so its better to preserve readability here